--- a/build/make/gen_msvs_vcxproj.sh
+++ b/build/make/gen_msvs_vcxproj.sh
@@ -435,6 +435,9 @@
             tag_content PreprocessorDefinitions "WIN32;$debug;_CRT_SECURE_NO_WARNINGS;_CRT_SECURE_NO_DEPRECATE$extradefines;%(PreprocessorDefinitions)"
             tag_content RuntimeLibrary $runtime
             tag_content WarningLevel Level3
+            if [ $plat = "Win64" ] &&  [ "x$W64OPT" != "x" ]; then
+                tag_content AdditionalOptions "$W64OPT"
+            fi
             if ${werror:-false}; then
                 tag_content TreatWarningAsError true
             fi
